src/coprocessor/dag/rpn_expr/impl_compare.rs

```diff
@@ -22,15 +21,29 @@ impl<C: Comparer> RpnFnCompare<C> {
             _phantom: std::marker::PhantomData,
         }
     }
+}
 
-    #[inline]
-    fn call(
-        _ctx: &mut EvalContext,
-        _payload: RpnFnCallPayload<'_>,
-        lhs: &Option<C::T>,
-        rhs: &Option<C::T>,
-    ) -> Result<Option<i64>> {
-        C::compare(lhs, rhs)
+impl<C: Comparer> RpnFunction for RpnFnCompare<C> {
+    fn name(&self) -> &'static str {
+        "RpnFnCompare"
+    }
+
+    fn args_len(&self) -> usize {
+        2
+    }
+
+    fn eval(
+        &self,
+        rows: usize,
+        ctx: &mut EvalContext,
+        payload: RpnFnCallPayload<'_>,
+    ) -> Result<VectorValue> {
+        let rpn_fn = compare_fn::<C>();
```

Comment:
Is this a temporary code? Is need to call `compare_fn` every time?

Reply:
It's temporary. As `RpnFunction` trait is removed in #4861, this is also removed.